networkx/generators/random_graphs.py

```diff
@@ -962,3 +963,79 @@ def random_powerlaw_tree_sequence(n, gamma=3, seed=None, tries=100):
           "Exceeded max (%d) attempts for a valid tree sequence."%tries)
     return False
 
+
+def random_kernel_graph(n, kernel_integral, kernel_root=None, seed=None):
+    """Return an random graph based on the specified kernel.
+
+    The algorithm chooses each of the `[n(n-1)]/2` possible edges with
+    probability specified by a kernel `\kappa(x,y)` [1]_.  The kernel 
+    `\kappa(x,y)` must be a symmetric (in x,y), non-negative, bounded function.
```

Comment:
x and y should be in math mode as well, since they refer to variables that are typeset previously in math mode.